### ISSUE TITLE
'toco_from_protos: not found' error found using prebuilt tensorflow binary for Raspbian Stretch 9 on Raspberry Pi 3 Model B

### ISSUE THREAD

**Issue by gragundier** [tensorflow/tensorflow#21198]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1. It must be a bug, a feature request, or a significant problem with documentation (for small docs fixes please send a PR instead).
2. The form below must be filled out.
3. It shouldn't be a TensorBoard issue. Those go [here](https://github.com/tensorflow/tensorboard/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: 
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Raspbian Stretch 9
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: Raspberry Pi 3 Model B
- **TensorFlow installed from (source or binary)**:binary
- **TensorFlow version (use command below)**:
tf.GIT_VERSION returns v1.9.0-0-g25c197e
- **Python version**:3.5.3
- **Bazel version (if compiling from source)**:N/A
- **GCC/Compiler version (if compiling from source)**:N/A
- **CUDA/cuDNN version**:N/A
- **GPU model and memory**:Please refer to Raspberry Pi 3 Model B specs, but I use the cpu only version of tensorflow.
- **Exact command to reproduce**:
use example code in tf-lite wiki... One sec...
# Converting a GraphDef from file. converter = lite.TocoConverter.from_frozen_graph( graph_def_file, input_arrays, output_arrays) tflite_model = converter.convert() open("converted_model.tflite", "wb").write(tflite_model)
This was taken from tf.contrib.lite.TocoConverter's API webpage.

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.
tl;dw2r run the script on the above specs and you get 'toco_from_protos: not found'
long story:
I am trying to convert a frozen pb file into a tflite format.  I have a Windows 10 system, but tflite isn't available yet on the Windows 10 binaries.  The only other system that I have available is a Raspberry Pi so I figured that its version of tensorflow should have access to the tflite segment of the api.  Sure enough, it was there.  But when I ran the script, the following error occurred.  I can try to use Ubuntu, but it's been a bit of a learning curve trying to manage all the systems I have built on my Windows computer with Linux alternatives.  If at all possible, I would like to have a solution on either of these devices.  

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by gragundier** [comment]
Whilst waiting for a response, I found this stack overflow answer helpful for my situation. https://stackoverflow.com/questions/50902067/how-to-import-the-tensorflow-lite-interpreter-in-python

**Comment by gragundier** [comment]
Ok, folks. Try using the python2.7 nightly build.  I think this is the only way we will get the interpreter to work atm.  For some reason, python3 doesn't play nice with the interpreter.

**Comment by gragundier** [comment]
To aselle:
Just a Raspberry Pi problem.  Interpreter works like a charm on x86_64 ubuntu.  Raspberry Pi throws out an error.  My python2 is broken on my Raspberry Pi at the moment, but trying the 2.7 build worked for someone else.  
To those looking for a python3 workaround:
If you really want python3 to work, thortex's fix seems to be working, but I've been running his script for 12 hrs now.  Furthermore, you need a usb drive as additional swap space.  https://github.com/thortex/rpi3-tensorflow

**Comment by gragundier** [comment]
TL;DR: Script mention earlier doesn't fix interpreter issue.  Interpreter and more specifically the interpreter wrapper seems to be the problem.  

Long story:
thortex's script "worked."  If I open a terminal, go:
`python3`
`import tensorflow as tf`
`dir(tf.contrib.lite)`
I get:
`['DecodeError', 'Interpreter', 'OpHint', 'PY3', 'TocoConverter', '__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__spec__', '_freeze_graph', '_graph_pb2', '_is_frozen_graph', '_session', '_text_format', 'absolute_import', 'constants', 'convert_op_hints_to_stubs', 'division', 'freeze_saved_model', 'get_tensors_from_tensor_names', 'global_variables_initializer', 'import_graph_def', 'print_function', 'set_tensor_shapes', 'signature_constants', 'tag_constants', 'tensor_name', 'tf_graph_util', 'toco_convert', 'toco_convert_protos']`
BUT as soon as I try freedomtan.py's [label_image.py](https://github.com/tensorflow/tensorflow/pull/19736/commits) script, I get something like:
`2018-08-03 04:58:17.421863: F tensorflow/core/framework/variant_op_registry.cc:104] Check failed: existing == nullptr (0x212ae74 vs. nullptr)Unary VariantDecodeFn for type_name: tensorflow::Tensor already registered
Aborted`
I enabled `tf.enable_eager_execution()` and ran the script on thonny:
::Start Thonny Copy Pasta::
%Run freedomtan.py
Backend terminated (returncode: -6)
Fatal Python error: Aborted

Current thread 0x76fdb010 (most recent call first):
  File "<frozen importlib._bootstrap>", line 222 in _call_with_frames_removed
  File "<frozen importlib._bootstrap_external>", line 914 in create_module
  File "<frozen importlib._bootstrap>", line 577 in module_from_spec
  File "<frozen importlib._bootstrap>", line 666 in _load_unlocked
  File "<frozen importlib._bootstrap>", line 693 in _load
  File "/usr/lib/python3.5/imp.py", line 342 in load_dynamic
  File "/usr/lib/python3.5/imp.py", line 242 in load_module
  File "/home/pi/.local/lib/python3.5/site-packages/tensorflow/contrib/lite/python/interpreter_wrapper/tensorflow_wrap_interpreter_wrapper.py", line 24 in swig_import_helper
  File "/home/pi/.local/lib/python3.5/site-packages/tensorflow/contrib/lite/python/interpreter_wrapper/tensorflow_wrap_interpreter_wrapper.py", line 28 in <module>
  File "<frozen importlib._bootstrap>", line 222 in _call_with_frames_removed
  File "<frozen importlib._bootstrap_external>", line 673 in exec_module
  File "<frozen importlib._bootstrap>", line 673 in _load_unlocked
  File "<frozen importlib._bootstrap>", line 958 in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 969 in _find_and_load
  File "<frozen importlib._bootstrap>", line 986 in _gcd_import
  File "/usr/lib/python3.5/importlib/__init__.py", line 126 in import_module
  File "/home/pi/.local/lib/python3.5/site-packages/tensorflow/python/util/lazy_loader.py", line 42 in _load
  File "/home/pi/.local/lib/python3.5/site-packages/tensorflow/python/util/lazy_loader.py", line 53 in __getattr__
  File "/home/pi/.local/lib/python3.5/site-packages/tensorflow/contrib/lite/python/interpreter.py", line 50 in __init__
  File "/home/pi/wontae-cv/freedomtan.py", line 65 in <module>
  File "/usr/lib/python3/dist-packages/thonny/shared/thonny/backend.py", line 588 in execute_source
  File "/usr/lib/python3/dist-packages/thonny/shared/thonny/backend.py", line 427 in _execute_source_ex
  File "/usr/lib/python3/dist-packages/thonny/shared/thonny/backend.py", line 374 in _execute_file
  File "/usr/lib/python3/dist-packages/thonny/shared/thonny/backend.py", line 155 in _cmd_Run
  File "/usr/lib/python3/dist-packages/thonny/shared/thonny/backend.py", line 119 in handle_command
  File "/usr/lib/python3/dist-packages/thonny/shared/thonny/backend.py", line 97 in mainloop
  File "/usr/lib/python3/dist-packages/thonny/shared/backend_launcher.py", line 41 in <module>
Resetting ...
>>> 
::End Thonny Copy Pasta::
I'm not completely comfortable with markdown, so I don't know how to format well.  Hope it's at least somewhat readable.  Anyways, the interpreter is most definitely the issue, specifically the wrapper.  
I'm going to rerun spec scripts again because I'm going to wipe my OS to see if I can fix my broken python/pip issue.
pi@pi:~/wontae-cv $ python3 -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"
unknown 1.9.0
[tf_env.txt](https://github.com/tensorflow/tensorflow/files/2256092/tf_env.txt)

**Comment by gragundier** [comment]
I wiped my raspberry pi clean.  Installed Raspbian Stretch 4.14.52-v7+ Lite.  Installed the nightly python2.7 build and received the same interpreter error.  I'd love to give you all the error log, but I have no idea how to do that from a command-line only os.

**Comment by gragundier** [comment]
It's broken on 1.8 cit.tensorflow.org package as well. 😢

**Comment by gragundier** [comment]
Ok, I've more or less figured it out.  The ARMv7 NEON implementations are interfering with the the interpreter wrapper.  Use the ci.tensorflow.org pi-zero implementations.  They work for both python3 and python 2.  It's all built on docker apparently, so just change the file names and install like any pip install.  To be explicit: 
cp34->cp35
armv6l->arvm7l that's an lowercase 'L' at the end there.

**Comment by gragundier** [comment]
If you need to compile from source for whatever reason.  Cross compile off of docker and DO NOT USE the armv7 settings.  Instead use something like PI_ONE.  
[It's all here](https://www.tensorflow.org/install/install_raspbian)
`$ tensorflow/tools/ci_build/ci_build.sh PI tensorflow/tools/ci_build/pi/build_raspberry_pi.sh PI_ONE`

**Comment by freedomtan** [comment]
@gragundier FYI, I was able to build pip package on RPI 3 w/ my one-line patch (https://github.com/tensorflow/tensorflow/pull/21109) and ran my label_image.py without problem.

```
bazel build --config opt \
--local_resources 1024.0,0.5,0.5 \
--cxxopt=-mfpu=neon-vfpv4 \
--cxxopt=-ftree-vectorize \
--cxxopt=-funsafe-math-optimizations \
--cxxopt=-ftree-loop-vectorize \
--cxxopt=-fomit-frame-pointer \
//tensorflow/tools/pip_package:build_pip_package
```

**Comment by tofulawrence** [comment]
@gragundier Did @freedomtan's patch work? If so, I'd like to close the issue. Thanks!

**Comment by tensorflowbutler** [comment]
Nagging Assignees @petewarden, @aselle: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by gragundier** [comment]
I haven't been able to test on the tinkerboard but it works for rapsberry Pi's closing!
